File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-12.cs
public class Solution {
    public int CharacterReplacement(string s, int k) {
        HashSet<char> set = new HashSet<char>(s);

        for(int i = 0; i < s.Length; i++){
            set.Add(s[i]);
        }
        int max = 0;
        foreach(char c in set){
            int count = 0;
            int l = 0;
            for(int r = 0; r < s.Length; r++){
                if(s[r] == c){
                    count++;
                }

                while((r - l) + 1 - count > k){
                    if(s[l] == c){
                        count--;
                    }
                    l++;
                }
                max = Math.Max(r - l + 1, max);
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/trapping-rain-water/submission-12.java
class Solution {
    public int trap(int[] height) {
     int l = 0;
     int r = height.length - 1;

     int lMax = 0;
     int rMax = 0;

     int res = 0;

     while(l < r){
        if(height[l] < height[r]){
            int sum = lMax - height[l];
            if(sum < 0){
                sum = 0;
            }
            res += sum;
            lMax = Math.max(lMax, height[l]);
            l++;
        }else{
            int sum = rMax - height[r];
            if(sum < 0){
                sum = 0;
            }

            res += sum;
            
            rMax = Math.max(rMax, height[r]);
            r--;

        }
     }
     return res;   
    }
}
